server/src/main/java/org/elasticsearch/common/settings/SettingsModule.java

```diff
@@ -176,12 +176,28 @@ private void registerSetting(Setting<?> setting) {
                     throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
                 }
                 nodeSettings.put(setting.getKey(), setting);
```

Comment:
not a big thing but this validation should be done before adding the setting into the map